libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookClientWrapper.cs

```diff
@@ -0,0 +1,162 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Bot.Builder.Adapters.Facebook.FacebookEvents;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Adapters.Facebook
+{
+    public class FacebookClientWrapper
+    {
+        private readonly FacebookAdapterOptions _options;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FacebookClientWrapper"/> class.
+        /// </summary>
+        /// <param name="options">An object containing API credentials, a webhook verification token and other options.</param>
+        public FacebookClientWrapper(FacebookAdapterOptions options)
+        {
+            _options = options ?? throw new ArgumentNullException(nameof(options));
+        }
+
+        /// <summary>
+        /// Call one of the Facebook APIs.
+        /// </summary>
+        /// <param name="path">Path to the API endpoint, for example `/me/messages`.</param>
+        /// <param name="payload">An object to be sent as parameters to the API call..</param>
+        /// <param name="method">HTTP method, for example POST, GET, DELETE or PUT.</param>
+        /// <param name="cancellationToken">A cancellation token for the task.</param>
+        /// <returns>A task representing the async operation.</returns>
+        public virtual async Task<string> SendMessageAsync(string path, FacebookMessage payload, HttpMethod method = null, CancellationToken cancellationToken = default)
+        {
+            if (path == null)
+            {
+                throw new ArgumentNullException(nameof(path));
+            }
+
+            if (payload == null)
+            {
+                throw new ArgumentNullException(nameof(payload));
+            }
+
+            var proof = GetAppSecretProof();
```

Comment:
I'm not sure this will be null. The `BitConverter.ToString(value)` method inside **GetAppSecretProof** should not return null I think, and should throw and exception if value is null, according to the documentation.